### ISSUE TITLE
[Documentation] Install page missing instructions for onnxruntime-rocm

### ISSUE THREAD

**Issue by shaunclimbon** [microsoft/onnxruntime#21036]
### Describe the documentation issue

The [installation matrix](https://onnxruntime.ai/) provides correct guidance for ORT with ROCm (i.e. build from source) but the actual install documentation is missing it.  In particular, the [Python Installs](https://github.com/microsoft/onnxruntime/blob/gh-pages/docs/install/index.md#python-installs) section covers ORT>CPU and ORT>GPU>CUDA but not ORT>GPU>ROCm.  Albeit Linux specific, it still seems like this info should be present.

As of reporting this, building the onnxruntime-rocm wheel was the recommended route (as documented [here](https://onnxruntime.ai/docs/build/eps.html#amd-rocm)) but it's worth noting that I found pre-built wheels in a couple locations:
- https://download.onnxruntime.ai/ (these seem to be for training with ORT)
- https://repo.radeon.com/rocm/manylinux/ (e.g. see 6.0.2 and 6.1 folders)

### Page / URL

https://onnxruntime.ai/docs/install/

**Comment by MaanavD** [comment]
Hey @shaunclimbon , do you have an email I could follow up with you on? I'm happy to add these missing instructions, but I want to make sure we accurately represent them in a way you're happy with :)

**Comment by shaunpamd** [comment]
Thanks for the prompt response MaanavD.  Yes, above suggestions looks good to me.  Mostly just trying to help users like me that miss the installation matrix.  And I figure the links to pre-built wheels could be useful to others that come across this.

**Comment by MaanavD** [comment]
Fixed in #21124

**Comment by NicholasFlamy** [comment]
Is building for a specific architecture documented anywhere? I am under the impression that to build for my RX 6700 XT, without building everything else and getting a 30 GB image, I need these environmental variables at build time:
```
AMDGPU=gfx1031
HSA_OVERRIDE=1
HSA_OVERRIDE_GFX_VERSION=10.3.0
```
